connectors/slack/template/storage.js

```diff
@@ -0,0 +1,48 @@
+const Sdk = require('@fusebit/add-on-sdk');
+
+let storage;
+module.exports.ensureStorage = (ctx) => {
+  if (!storage) {
+    storage = Sdk.getStorageClient(ctx);
+  }
+};
+
+module.exports.teamToHandler = null;
+module.exports.tokenToTeam = null;
+let mappingTimestamp;
+module.exports.get = async () => {
+  let result = await storage.get();
+  module.exports.teamToHandler = (result && result.teamToHandler) || {};
+  module.exports.tokenToTeam = (result && result.tokenToTeam) || {};
+  mappingTimestamp = Date.now();
+};
+
+module.exports.put = async (applyChanges) => {
+  while (true) {
+    // TODO Maybe add some backoff?
+
+    applyChanges();
+
+    try {
+      await storage.put({
+        teamToHandler: module.exports.teamToHandler,
+        tokenToTeam: module.exports.tokenToTeam,
+      });
+      break;
+    } catch (e) {
+      Sdk.debug('Cache conflict');
+      if (e.statusCode && e.statusCode == 409) {
+        // Local copy is stale, need to refresh
+        await module.exports.get();
+      } else throw e;
+    }
+  }
+};
+
+module.exports.ensureCache = async () => {
+  // If no in-memory cache present, or the cache is more than 15 sec old,
+  // reload from storage
```

Comment:
The 15 sec is completely arbitrary. Can we just leave it be and cause a conflict on put?

Reply:
I'm not quite sure how Lambda instancing works, but what if a registration happens in the background and there is no PUT happening in an instance for a while? We risk a newly-created function not getting its events for multiple minutes.

Reply:
You are right